handlers/api/commentApiHandler.js

```diff
@@ -19,6 +19,24 @@ CommentApiHandler.addComment = function (req, res) {
         });
 };
 
+CommentApiHandler.updateComment = function (req, res) {
+    comments.update(
+        sanitizer(req.body.commentId),
+        sanitizer(req.body.comment),
+        sanitizer(req.body.software_version_id),
+
+        function (result, error) {
+            comments.getById(sanitizer(req.body.commentId), function(comment) {
+                // if not the owner of this comment and not an admin
```

Comment:
shouldn't we fetch the comment by ID before updating, check whether the user has permissions, and then call comments.update ? If I understand correctly right now we show a user an error message but the comment is updated anyway?